src/error.rs

```diff
@@ -239,91 +236,69 @@ impl Clone for Error {
 
 #[cfg(feature = "security")]
 fn from_sslerror_ref(err: &ssl::Error) -> ErrorKind {
-    match err {
-        &SslError::ZeroReturn => ErrorKind::Ssl(SslError::ZeroReturn),
-        &SslError::WantRead(ref e) => ErrorKind::Ssl(SslError::WantRead(clone_ioe(e))),
-        &SslError::WantWrite(ref e) => ErrorKind::Ssl(SslError::WantWrite(clone_ioe(e))),
-        &SslError::WantX509Lookup => ErrorKind::Ssl(SslError::WantX509Lookup),
-        &SslError::Stream(ref e) => ErrorKind::Ssl(SslError::Stream(clone_ioe(e))),
-        &SslError::Ssl(ref es) => ErrorKind::Ssl(SslError::Ssl(es.clone())),
+    if let Some(io_err) = err.io_error() {
+        return ErrorKind::Io(clone_ioe(io_err));
+    }
+    if let Some(ssl_err) = err.ssl_error() {
+        return ErrorKind::Ssl(From::from(ssl_err.to_owned()));
     }
+
+    // xxx: return something as default?
+    unreachable!()
```

Comment:
Strange... this almost seems like a step back in terms of API soundness from openssl. How unfortunate.